app/models/edition/custom_lead_image.rb

```diff
@@ -4,14 +4,11 @@ module Edition::CustomLeadImage
   included do
     validates :body,
               format: {
-                without: /^!!1([^\d]|$)/,
-                message: "cannot have a reference to the first image in the text",
+                without: ->(edition) { /^!!#{edition.lead_image_index + 1}([^\d]|$)|\[Image: #{edition.lead_image.filename}\]/ },
```

Comment:
Or alternatively, I wonder if it could be refactored into a custom validator method or class, rather than being defined as a 'format' rule?

That might make more 'room' to expand it into something more readable.

Reply:
I should also point out that the `[Image:]` syntax allows any amount of whitespace (including none) before the filename.

For example, these are all equivalent:
- `[Image:example.jpg]`
- `[Image: example.jpg]`
- `[Image:    example.jpg]`

So it'd be good to have test coverage for this (and then the regex will need updating accordingly).

For what it's worth, [the regex used by Govspeak](https://github.com/alphagov/govspeak/blob/ae8136a9bba22ed55479974544fba27c2f98bd0a/lib/govspeak.rb#L356C24-L356C48) is:

```ruby
/^\[Image:\s*(.*?)\s*\]/
```

Where `\s` whitespace is allowed `*` times (zero or more).

Also, we should really make use of `Regex.escape()` for the image filename. Otherwise the `.` in the file extension will technically match _any character_, not just a literal `.` fullstop/period.

So something like:

```ruby
/\[Image: #{Regex.escape(edition.lead_image.filename)}\]/
```

Reply:
i've I've pulled the validation into it's own method now in this commit https://github.com/alphagov/whitehall/pull/8433/commits/71207de43df034120ee0f35ad9d657caad7269b lmn what you think

Reply:
I think we're okay on this front now but let me know if that's not the case